R/layer.R

```diff
@@ -963,3 +965,17 @@ cleanup_mismatched_data <- function(data, n, fun) {
   data[failed] <- NULL
   data
 }
+
+normalise_label <- function(label) {
+  if (is.null(label)) {
+    return(NULL)
+  }
+  if (obj_is_list(label)) {
+    label[lengths(label) == 0] <- ""
+    labels <- lapply(labels, `[`, i)
+  }
+  if (is.expression(label)) {
+    label <- unclass(as.list(label))
```

Comment:
The `unclass()` idea is from #6639.
It counters length 1 expression subclasses, which is admittedly a niche problem.